text/0000-remove-shell-specific-processes.md

```diff
@@ -0,0 +1,204 @@
+
+
+# How it Works
+[how-it-works]: #how-it-works
+
+## Example 1 - A Shell Process
+The Paketo .Net Execute Buildpack may generates shell processes similar to the following:
+```
+[[processes]]
+type = "web"
+command = "dotnet my-app.dll --url http://0.0.0.0:${PORT:-8080}"
+direct = false
+```
+NOTE: the buildpack API used by this buildpack (`0.5`) predates the introduction of `default`
+
+Using the new API this process could look like:
+```
+[[processes]]
+type = "web"
+command = ["dotnet", "my-app.dll", "--urls", "http://0.0.0.0:${PORT}"] # the default value of PORT would need to be provided in a layer
+default = true
+```
+Things to note:
+* In the above example I have eliminated the dependency on `cmd.exe` instead of explicitly adding it to the command, because it is likely unnecessary (folks who actually use .Net core will surely correct me if I am wrong).
+* If the buildpack authors believed that `--urls` should be overridable they could set move the last two arguments from `command` to `args`.
+
+## Example 2 - A Script Process
+When a buildpack upgrade to the new buildpack API, it must convert any existing shell processes into direct processes.
+
+The [Paketo Yarn Start Buildpack](https://github.com/paketo-buildpacks/yarn-start) currently may generate a script processes similar to the following:
+
+```
+[[processes]]
+type = "web"
+command = "pre-start.sh && nodejs server.js && post-start.sh"
+direct = false
+default = false
+```
+
+Using the new API this process look like:
+```
+[[processes]]
+type = "web"
+command = ["bash", "-c", "pre-start.sh && nodejs server.js && post-start.sh"]
+default = false
+```
+
+## What About Profile Scripts?
+
+### `<layer>/profile.d/*`
+When a buildpack upgrades to the new buildpack API, buildpack-provided profile scripts will no longer be supported. Instead buildpacks can use `exec.d`. Most existing profiles scripts can be easily converted.
+
+**Example 1: **
+
+**Example 2: **
+
+### `<app>/.profile`
+
+When a platform upgrades to the new platform API, user-provided profile scripts will no longer be supported. A buildpack could provide an identical or similar interface by detecting a `.profile` file in the app dir, wrapping it so that it implements the `exec.d` interface and add creating an executable with that layer. A platform that wishes to always provide that functionality could add a buildpack like this to every build. By making dynamic runtime environment modification a buildpack concern and not duplicating similar behavior in the platform API, we reduce the surface area and complexity of the platform API and thus improve normalization of the spec.
```

Comment:
> A platform that wishes to always provide that functionality could add a buildpack like this to every build.

I recall something similar came up in discussions around prepare phase & project.toml (using buildpacks to perform actions instead of the platform - e.g., setting build time env) and I wonder if maybe that idea deserves its own RFC. It seems like a sticking point in these conversations has been "who is responsible" for creating these generic buildpacks, how would they be supported & distributed, etc.

Reply:
@natalieparellano I agree that an RFC for this sounds like a good idea.

If we, as a project, wanted to provide buildpacks to make it easy for platforms or buildpck ecosystems to adopts extensions or conventions, that would deserve its own RFC. However, I don't think we need to block this RFC on that. Whether or not we take the extra step of providing common buildpacks (and I suspect we should) an app->lifecycle API is fundamentally untenable and I would argue that we need to remove this feature either way.

Reply:
@natalieparellano can this comment be resolved given the discussion has moved to https://github.com/buildpacks/rfcs/pull/175? Or do you feel that this RFC need further language to address this before approval?